tcs/csp_solver_two_tank_tes.cpp

```diff
@@ -1138,231 +1554,188 @@ bool C_csp_two_tank_tes::charge(double timestep /*s*/, double T_amb /*K*/, doubl
+  //              }
+  //              else {
+  //                  m_dot_tank_guess_1 = fmin(m_dot_tank_upper, m_dot_tank_guess_1 + 0.01*(m_dot_tank_upper - m_dot_tank_lower));
+  //              }
+  //          }
+  //      }
+  //      else if (m_dot_tank_guess_1 == 0 || m_dot_tank_guess_2 == 0) {
+  //          // Try a 0 guess
+  //          double m_dot_bal = std::numeric_limits<double>::quiet_NaN();
+  //          int m_dot_bal_code = c_tes_chrg_solver.test_member_function(0., &m_dot_bal);
+
+  //          if (m_dot_bal < 1.E-3) {
+  //              m_dot_tank = 0.;
+  //              m_dot_solved = true;
+  //          }
+  //          else {
+  //              // Adjust 0 guess to avoid divide by 0 errors
+  //              m_dot_tank_guess_2 = fmax(m_dot_tank_guess_1, m_dot_tank_guess_2);
+  //              m_dot_tank_guess_1 = 1.e-3;
+  //          }
+  //      }
+
+  //      if (!m_dot_solved) {
+  //          // Solve for required tank mass flow
+  //          double tol_solved;
+  //          tol_solved = std::numeric_limits<double>::quiet_NaN();
+  //          int iter_solved = -1;
+		//	int m_dot_bal_solver_code = 0;
+
+  //          try
+  //          {
+  //              m_dot_bal_solver_code = c_tes_chrg_solver.solve(m_dot_tank_guess_1, m_dot_tank_guess_2, -1.E-3, m_dot_tank, tol_solved, iter_solved);
+  //          }
+  //          catch (C_csp_exception)
+  //          {
+  //              throw(C_csp_exception("Failed to find a solution for the cold tank mass flow"));
+  //          }
+
+		//	if (m_dot_bal_solver_code != 3)
+		//	{
+		//		double abc = 1.23;
+		//	}
+
+  //          if (std::isnan(m_dot_tank)) {
+  //              throw(C_csp_exception("Failed to converge on a valid tank mass flow."));
+  //          }
+  //      }
+		//else
+		//{
+		//	double abcde = 1.23;
+		//}
 
-        // The other needed outputs are not all saved in member variables so recalculate here
-        mc_cold_tank.energy_balance(timestep, 0.0, m_dot_tank, 0.0, T_amb, T_cold_ave, q_heater_cold, q_dot_loss_cold);
+  //      // The other needed outputs are not all saved in member variables so recalculate here
+  //      mc_cold_tank.energy_balance(timestep, 0.0, m_dot_tank, 0.0, T_amb, T_cold_ave, q_heater_cold, q_dot_loss_cold);
 
-        double eff, q_trans;
-        eff = q_trans = std::numeric_limits<double>::quiet_NaN();
-        mc_hx.hx_charge_mdot_field(T_field_hot_in, m_dot_field, T_cold_ave,
-            eff, T_field_cold_out, T_hot_tank_in, q_trans, m_dot_tank);
+  //      double eff, q_trans;
+  //      eff = q_trans = std::numeric_limits<double>::quiet_NaN();
+  //      mc_hx.hx_charge_mdot_fieldabc(T_field_hot_in, m_dot_field, T_cold_ave,
+  //          eff, T_field_cold_out, T_hot_tank_in, q_trans, m_dot_tank);
+
+  //      mc_hot_tank.energy_balance(timestep, m_dot_tank, 0.0, T_hot_tank_in, T_amb, T_hot_ave, q_heater_hot, q_dot_loss_hot);
```

Comment:
either remove or make into a new HX child

Reply:
Prefer to keep old code around until people have exercised the new controller for a while